### PR TITLE
feat: Create new `HorizontalPage` layout and use it for ubuntu_bootstrap

### PR DESCRIPTION
**Pull request by spydon**
[Screencast from 2024-01-12 14:00:45.webm](https://github.com/canonical/ubuntu-desktop-provision/assets/744771/98c17436-2799-49f5-9248-c3bf6dcd94ae)

### CODE REVIEW

File: packages/ubuntu_provision/test/network/network_page_test.dart [hunk 1]
```diff
@@ -72,6 +74,7 @@ void main() {
   });
 
   testWidgets('selects wifi', (tester) async {
+    await tester.binding.setSurfaceSize(const Size(800, 800));
```
Comment (d-loose) [inline packages/ubuntu_provision/test/network/network_page_test.dart:77]: Could we add a TODO here and/or on the network page to keep track of the places that will need some polishing?

File: packages/ubuntu_bootstrap/test/installer_wizard_test.dart [hunk 2]
```diff
@@ -88,6 +88,7 @@ void main() {
   });
 
   testWidgets('guided reformat', (tester) async {
+    await tester.binding.setSurfaceSize(const Size(800, 800));
```
Comment (d-loose) [inline packages/ubuntu_bootstrap/test/installer_wizard_test.dart:91]: Same here, it'd be great to document which pages break or have overflows :)

File: packages/ubuntu_bootstrap/integration_test/ubuntu_bootstrap_test.dart [hunk 3]
```diff
@@ -179,6 +179,7 @@ void main() {
   });
 
   testWidgets('tpm', (tester) async {
+    await tester.binding.setSurfaceSize(const Size(1000, 1000));
```
Comment (d-loose) [inline packages/ubuntu_bootstrap/integration_test/ubuntu_bootstrap_test.dart:182]: Same here, if possible

**Review by d-loose — APPROVED** [review]
A few minor comments regarding documenting the small screen issues. Otherwise, once the test failure related to the build context extension is fixed - LGTM :+1:

Great work - it's really nice to see @elioqoshi's and @anasereijo's designs come to life!

File: packages/ubuntu_provision/test/network/network_page_test.dart (hunk 1 above)
Comment (spydon) [reply packages/ubuntu_provision/test/network/network_page_test.dart:78]: Done!

File: packages/ubuntu_bootstrap/integration_test/ubuntu_bootstrap_test.dart [hunk 4]
```diff
@@ -208,31 +209,31 @@ void main() {
...
+    //final windowClosed = YaruTestWindow.waitForClosed();
+    //await tester.tapContinueTesting();
+    //await expectLater(windowClosed, completes);
 
-    await verifySubiquityConfig(
-      capability: GuidedCapability.CORE_BOOT_ENCRYPTED,
-    );
+    //await verifySubiquityConfig(
+    //  capability: GuidedCapability.CORE_BOOT_ENCRYPTED,
+    //);
```
Comment (d-loose) [inline packages/ubuntu_bootstrap/integration_test/ubuntu_bootstrap_test.dart:236]: I suspect you didn't mean to commit this

Comment (spydon) [reply packages/ubuntu_bootstrap/integration_test/ubuntu_bootstrap_test.dart]: Hahah ooops, fixed now!

**Pull request merged**